source/server/options_impl.cc

```diff
@@ -160,6 +160,12 @@ OptionsImpl::OptionsImpl(std::vector<std::string> args,
                                                   "Comma-separated list of extensions to disable",
                                                   false, "", "string", cmd);
 
+  TCLAP::ValueArg<std::string> socket_path("", "socket-path", "Path to hot restart socket file",
+                                           false, "@envoy_domain_socket", "string", cmd);
+
+  TCLAP::ValueArg<mode_t> socket_mode("", "socket-mode", "Socket file permission", false, S_IRWXU,
```

Comment:
Deleted octalToDecimal

Reply:
I think  socket_mode must be of mode_t type because for string type if the input is other than an integer will be accepted which will not desirable.WDYT?

Reply:
The return value of `atoull` tells you if it was successfully converted to a number. If it fails, throw a `MalformedArgvException`